app/javascript/packages/address-search/components/full-address-search.spec.tsx

```diff
@@ -320,4 +320,57 @@ describe('FullAddressSearch', () => {
       await expect(handleLocationsFound).to.eventually.be.called();
     });
   });
+
+  context('Address Search with Results Section Heading', () => {
+    let server: SetupServer;
+    before(() => {
+      server = setupServer(
+        http.post(locationsURL, () => HttpResponse.json([{ name: 'Baltimore' }])),
+      );
+      server.listen();
+    });
+
+    after(() => {
+      server.close();
+    });
+
+    it('renders the results section heading when passed in', async () => {
+      const handleLocationsFound = sandbox.stub();
+      const onSelect = sinon.stub();
+      const resultsSectionHeadingText = 'Mock Heading';
+      const { findByText, getByLabelText, getByText } = render(
+        <SWRConfig value={{ provider: () => new Map() }}>
+          <FullAddressSearch
+            usStatesTerritories={usStatesTerritories}
+            onFoundLocations={handleLocationsFound}
+            locationsURL={locationsURL}
+            registerField={() => undefined}
+            handleLocationSelect={onSelect}
+            disabled={false}
+            resultsSectionHeading={() => <h2>{resultsSectionHeadingText}</h2>}
+          />
+        </SWRConfig>,
+      );
+
+      await userEvent.type(
+        getByLabelText('in_person_proofing.body.location.po_search.address_label'),
+        '200 main',
+      );
+      await userEvent.type(
+        getByLabelText('in_person_proofing.body.location.po_search.city_label'),
+        'Endeavor',
+      );
+      await userEvent.selectOptions(
+        getByLabelText('in_person_proofing.body.location.po_search.state_label'),
+        'DE',
+      );
+      await userEvent.type(
+        getByLabelText('in_person_proofing.body.location.po_search.zipcode_label'),
+        '17201',
+      );
+      await userEvent.click(getByText('in_person_proofing.body.location.po_search.search_button'));
+
+      await findByText(resultsSectionHeadingText);
```

Comment:
I would expect to see assertion statements in your testing.

Reply:
Technically `findBy*` and `getBy*` can be assertions on their own, since they will throw errors if the text isn't found. That being said, I personally tend to throw in a `.to.exist();` just to be extra explicit.

```suggestion
      expect(await findByText(resultsSectionHeadingText)).to.exist();
```

Reply:
True! That is why I suggested it. Although we are testing all of the above getBy.. - this is why the test was written - to check for this.